tests/ui/argument-suggestions/extra_arguments.stderr

```diff
@@ -97,7 +97,7 @@ note: function defined here
   --> $DIR/extra_arguments.rs:3:4
    |
 LL | fn two_arg_same(_a: i32, _b: i32) {}
-   |    ^^^^^^^^^^^^ -------  -------
```

Comment:
Where did this span go? This seems wrong lol.

Reply:
The code marking those spans just marks the span of all arguments that aren't generics used in multiple positions. It's also not applied to anything without a body (e.g. foreign functions).

The message as a whole is: https://github.com/rust-lang/rust/blob/7551b1dc92636b04e76d17719493c902bc0cb6e2/tests/ui/argument-suggestions/extra_arguments.stderr#L90-L105

Why are the arguments even being given spans here? The message is noting where the definition is and it's a pile of spans. 

I can remove the third commit. It wasn't really the main point anyways and is the cause of the hundred files worth of changes.

Reply:
Going back looks like df93364 actually broke the span marking here. It used to mark only relevant parameters before the generic arg matching code was added.

Reply:
If you want to author a fix, I'm happy to review that too in a separate PR. But I think removing all non-generic spans is kinda orthogonal and unnecessary here.

Reply:
Reverted now.